src/linear_algebra/projection.lean

```diff
@@ -178,6 +178,14 @@ lemma exists_unique_add_of_is_compl (hc : is_compl p q) (x : E) :
 let ⟨u, hu₁, hu₂⟩ := exists_unique_add_of_is_compl_prod hc x in
   ⟨u.1, u.2, hu₁, λ r s hrs, prod.eq_iff_fst_eq_snd_eq.1 (hu₂ ⟨r, s⟩ hrs)⟩
 
+lemma eq_linear_proj_add_linear_proj_of_is_compl (hpq : is_compl p q) (x : E) :
+  x = p.linear_proj_of_is_compl q hpq x + q.linear_proj_of_is_compl p hpq.symm x :=
+begin
+  rcases exists_unique_add_of_is_compl hpq x with ⟨u, v, rfl, -⟩,
+  simp only [ map_add, add_zero, linear_proj_of_is_compl_apply_right,
+              zero_add, linear_proj_of_is_compl_apply_left ],
+end
```

Comment:
If you add these:
```lean
-- belongs in linear_algebra.prod
/-- `add_equiv.prod_comm` is also a `linear_equiv`. -/
@[simps apply]
def linear_equiv.prod_comm (R M N : Type*) [semiring R] [add_comm_monoid M] [add_comm_monoid N]
  [module R M] [module R N] : (M × N) ≃ₗ[R] (N × M) :=
{ to_fun := prod.swap,
  map_smul' := λ r ⟨m, n⟩, rfl,
  ..add_equiv.prod_comm }

-- belongs a little further up the file
@[simp]
lemma prod_comm_trans_prod_equiv_of_is_compl (h : is_compl p q) :
  linear_equiv.prod_comm R q p ≪≫ₗ prod_equiv_of_is_compl p q h =
    prod_equiv_of_is_compl q p h.symm :=
linear_equiv.ext $ λ _, add_comm _ _
```
then you can also prove this as
```suggestion
lemma eq_linear_proj_add_linear_proj_of_is_compl (hpq : is_compl p q) (x : E) :
  x = p.linear_proj_of_is_compl q hpq x + q.linear_proj_of_is_compl p hpq.symm x :=
begin
  dunfold linear_proj_of_is_compl,
  rw ←prod_comm_trans_prod_equiv_of_is_compl _ _ hpq,
  exact ((prod_equiv_of_is_compl _ _ hpq).apply_symm_apply x).symm,
end
```

Reply:
Although I'd argue the lemma should be stated in the opposite direction, which also saves the `(...).symm` from the proof above.